library/src/main/java/com/datatorrent/lib/io/jms/JMSTransactionableStore.java

```diff
@@ -229,6 +233,7 @@ public boolean isConnected()
 
   private String getQueueName(String appId, int operatorId)
   {
-    return appId + "-" + operatorId;
+    return getSubject() + "." + "metadata";
```

Comment:
For now it is okay but it is preferable to just store the queue name instead of composing it in getQueueName.